akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -152,6 +152,14 @@ class Flow[-In, +Out](delegate: scaladsl.Flow[In, Out]) {
    * downstream may run in parallel and may complete in any order, but the elements that
```

Comment:
The Stage itself can define it by overriding the `decide` and `restart` methods. I was thinking about this, and to give custom stages the same capabilities as our internal stages the factory function of `transform` must be something like `GeneralFlowMaterializerSettings => Stage` or `OperationAttributes => Stage`. We can have that in mind, but for now it is out of scope.